### PR TITLE
Iota

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: packages/Search/test/tstDetailsUtils.cpp [hunk 1]
```diff
@@ -12,8 +12,29 @@
...
 #include <algorithm>
+#include <numeric>
 #include <vector>
 
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DetailsUtils, iota, DeviceType )
+{
+    int const n = 10;
+    double const val = 3.;
+    Kokkos::View<double *, DeviceType> v( "v", n );
+    DataTransferKit::iota( v, 3. );
```
Comment (aprokop) [inline packages/Search/test/tstDetailsUtils.cpp]: Use `val` instead of 3.

**Comment by dalg24** [comment]
@aprokop 
>I think that
> 1. iota should be KOKKOS_FUNCTION
> 2. It should not call Kokkos::fence or otherwise sync.

I disagree. `iota()`, as well as `exclusivefPrefixSum()` and `lastElement()` in the same header file, was meant to be called from the host.

File: packages/Search/test/tstDetailsUtils.cpp (hunk 1 above)
Comment (dalg24) [reply packages/Search/test/tstDetailsUtils.cpp]: Good catch :)

**Comment by aprokop** [comment]
OK. But even if it's called from host, there is not need to include `fence` in it.

**Comment by dalg24** [comment]
So would you want to remove the `fence()` from `exclusivePrefixSum()` as well?
I am not convinced at this time, I would rather have it and consider removing it in the future.

**Comment by Rombur** [comment]
Can you change the message of the second commit `for_parallel()` -> `parallel_for()`

**Comment by dalg24** [comment]
@Rombur Done

**Pull request merged**